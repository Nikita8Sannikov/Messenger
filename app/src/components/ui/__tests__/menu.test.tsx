import { fireEvent, render, screen } from "@testing-library/react";

import MenuComponent from "@components/ui/menu";

describe("MenuComponent", () => {
	const mockProps = {
		children: <div>Test Child</div>,
		anchorEl: document.createElement("div"),
		anchorOrigin: {
			vertical: "top" as const,
			horizontal: "left" as const,
		},
		open: true,
		autoFocus: false,
		onClose: jest.fn(),
	};

	beforeEach(() => {
		jest.clearAllMocks();
	});

	it("should render when open is true", () => {
		render(<MenuComponent {...mockProps} />);
        
		const menu = screen.getByRole("menu");
		expect(menu).toBeInTheDocument();
		expect(menu).toHaveTextContent("Test Child");
	});

	it("should not render when open is false", () => {
		render(<MenuComponent {...mockProps} open={false} />);
        
		const menu = screen.queryByRole("menu");
		expect(menu).not.toBeInTheDocument();
	});

	it("should call onClose when backdrop is clicked", () => {
		render(<MenuComponent {...mockProps} />);
        
		const backdrop = document.querySelector(".MuiBackdrop-root");
		if (backdrop) {
			fireEvent.click(backdrop);
		}
        
		expect(mockProps.onClose).toHaveBeenCalledWith(expect.anything(), "backdropClick");
	});

	it("should call onClose when escape key is pressed", () => {
		render(<MenuComponent {...mockProps} />);
        
		const menuPaper = document.querySelector(".MuiMenu-paper");
		if (menuPaper) {
			fireEvent.keyDown(menuPaper, { key: "Escape" });
		}
        
		expect(mockProps.onClose).toHaveBeenCalledWith(expect.anything(), "escapeKeyDown");
	});

	it("should not autoFocus when autoFocus is false", () => {
		render(
			<MenuComponent 
				{...mockProps} 
				autoFocus={false}
			/>,
		);
        
		const menuPaper = document.querySelector(".MuiMenu-paper");
		expect(menuPaper).toHaveAttribute("tabindex", "-1");
	});

	it("should autoFocus when autoFocus is true", () => {
		render(
			<MenuComponent 
				{...mockProps} 
				autoFocus={true}
			/>,
		);
        
		const menuPaper = document.querySelector(".MuiMenu-paper");
		expect(menuPaper).toHaveAttribute("tabindex", "-1");
	});

	it("matches snapshot", () => {
		const { asFragment } = render(<MenuComponent {...mockProps} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
