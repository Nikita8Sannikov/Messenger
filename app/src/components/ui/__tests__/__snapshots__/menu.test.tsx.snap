// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuComponent matches snapshot 1`] = `<DocumentFragment />`;
